azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/legacy/appservice/AppServiceUtils.java

```diff
@@ -5,111 +5,36 @@
-import com.microsoft.azure.management.appservice.RuntimeStack;
-import com.microsoft.azure.management.appservice.WebAppBase;
-import com.microsoft.azure.toolkit.lib.common.exception.AzureExecutionException;
-import com.microsoft.azure.toolkit.lib.common.logging.Log;
+import com.microsoft.azure.toolkit.lib.appservice.model.OperatingSystem;
+import com.microsoft.azure.toolkit.lib.appservice.model.PricingTier;
+import com.microsoft.azure.toolkit.lib.appservice.service.IWebApp;
 import org.apache.commons.lang3.StringUtils;
 
-import java.lang.reflect.Field;
-import java.lang.reflect.Modifier;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.UUID;
-import java.util.regex.Pattern;
+import java.util.Objects;
 
 public class AppServiceUtils {
-
-    private static final String SERVICE_PLAN_NOT_FOUND = "Failed to get App Service Plan.";
-    private static final String UPDATE_APP_SERVICE_PLAN = "Updating App Service Plan...";
-    private static final List<PricingTier> pricingTiers = new ArrayList<>();
-
-    static {
-        // Init runtimeStack list
-        for (final Field field : PricingTier.class.getFields()) {
-            if (Modifier.isStatic(field.getModifiers())) {
-                try {
-                    pricingTiers.add((PricingTier) field.get(null));
-                } catch (IllegalAccessException e) {
-                    e.printStackTrace();
-                }
-            }
-        }
-    }
-
-    public static AppServicePlan getAppServicePlan(final String servicePlanName, final Azure azureClient,
-                                                   final String resourceGroup, final String servicePlanResourceGroup) {
-        if (StringUtils.isNotEmpty(servicePlanName)) {
-            final String servicePlanResGrp = getAppServicePlanResourceGroup(resourceGroup, servicePlanResourceGroup);
-            return azureClient.appServices().appServicePlans()
-                .getByResourceGroup(servicePlanResGrp, servicePlanName);
-        }
-        return null;
-    }
-
-    public static String getAppServicePlanResourceGroup(final String resourceGroup, final String appServicePlanResGrp) {
-        return StringUtils.isEmpty(appServicePlanResGrp) ? resourceGroup : appServicePlanResGrp;
-    }
-
-    public static String getAppServicePlanName(final String servicePlanName) {
-        return StringUtils.isEmpty(servicePlanName) ? generateRandomServicePlanName() : servicePlanName;
-    }
-
-    private static String generateRandomServicePlanName() {
-        return "ServicePlan" + UUID.randomUUID().toString().substring(0, 18);
-    }
-
     public static PricingTier getPricingTierFromString(final String pricingTierString) {
-        for (final PricingTier pricingTier : pricingTiers) {
-            if (pricingTier.toSkuDescription().size().equalsIgnoreCase(pricingTierString)) {
-                return pricingTier;
-            }
-        }
-        return null;
+        return PricingTier.fromString(pricingTierString);
```

Comment:
remove this method.

Reply:
this PR is not for this kind of refactor